src/main/java/File/FileManager.java

```diff
@@ -0,0 +1,222 @@
+     */
+    private static boolean getDone(String s) throws ErrorExceptions {
+        Scanner sc = new Scanner(s);
+        try {
+            String current = sc.next();
+            char done = current.charAt(4);
+            String d = "";
+            d = d + done;
+            if (d.equals("O")) {
+                return true;
+            } else if (d.equals("X")) {
+                return false;
+            } else {
+                throw new ErrorExceptions("Error: Cannot determine if item is done, file might be corrupted!");
+            }
+        } catch (NoSuchElementException e) {
+            throw new ErrorExceptions("Failed to load saved file (Completed?)" + System.lineSeparator()
+                    + e);
+        }
+    }
+
+    /**
+     * Reads the local save file and convert into task to add to list.
+     *
+     * @param f saved file.
+     * @param store ArrayList to store the tasks.
+     */
+    public static void read(File f, ArrayList<task> store) {
+        try {
+            Scanner sc = new Scanner(f);
+            while (sc.hasNext()) {
+                String current = sc.nextLine();
+                try {
+                    int type = FileManager.getType(current);
+                    String name = FileManager.getName(current);
+                    boolean done = FileManager.getDone(current);
+                    if (type == 1) {
+                        Todo t = new Todo(name, "[T]");
+                        if (done) {
+                            t.setDone();
+                        }
+                        store.add(t);
+                    } else if (type == 2) {
+                        try {
+                            String date = FileManager.getDate(current);
+                            Deadline d = new Deadline(name, "[D]");
+                            if (done) {
+                                d.setDone();
+                            }
+                            DateTimeManager.addDate(d, date);
+                            store.add(d);
+                        } catch (ErrorExceptions e) {
+                            System.out.println(e);
+                        }
+                    } else {
+                        try {
+                            String date = FileManager.getDate(current);
+                            Event e = new Event(name, "[E]");
+                            if (done) {
+                                e.setDone();
+                            }
+                            DateTimeManager.addDate(e, date);
+                            store.add(e);
+                        } catch (ErrorExceptions e) {
+                            System.out.println(e);
+                        }
+                    }
+                } catch (ErrorExceptions e) {
+                    System.out.println(e);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File does not exist!");
+        }
+    }
+}
```

Comment:
same as before! minimise your try-catch coverage to error-prone areas and use switch statements~